src/api/api_client.rs

```diff
@@ -111,28 +111,93 @@ where
 	Client: RpcClient,
 	Params: ExtrinsicParams<Index, Hash>,
 {
-	pub signer: Option<P>,
-	pub genesis_hash: Hash,
-	pub metadata: Metadata,
-	pub runtime_version: RuntimeVersion,
-	pub client: Client,
-	pub extrinsic_params_builder: Option<Params::OtherParams>,
+	signer: Option<P>,
+	genesis_hash: Hash,
+	metadata: Metadata,
+	runtime_version: RuntimeVersion,
+	client: Client,
+	extrinsic_params_builder: Option<Params::OtherParams>,
 }
 
+/// Setter and getter calls to the local cache, no substrate node calls involved.
 impl<P, Client, Params> Api<P, Client, Params>
 where
 	P: Pair,
-	MultiSignature: From<P::Signature>,
 	MultiSigner: From<P::Public>,
 	Client: RpcClient,
 	Params: ExtrinsicParams<Index, Hash>,
 {
+	/// Set the api signer account.
+	pub fn set_signer(&mut self, signer: P) {
+		self.signer = Some(signer);
+	}
+
+	/// Get the public part of the api signer account.
 	pub fn signer_account(&self) -> Option<AccountId> {
 		let pair = self.signer.as_ref()?;
 		let multi_signer = MultiSigner::from(pair.public());
 		Some(multi_signer.into_account())
 	}
 
+	/// Get the private key pair of the api signer.
+	pub fn signer(&self) -> Option<&P> {
+		match &self.signer {
+			Some(signer) => Some(signer),
+			None => None,
+		}
```

Comment:
This can certainly be simplified, no?

Reply:
Ah well, yeah. Forgot about the as_ref()